freedomofpress/securedrop
# Add updater GUI

## Status

Ready for review

## Description of Changes

Fixes #3076.

Changes proposed in this pull request:
 - Adds the Updater GUI and enables automatic launching of the updater when updates are needed in the Tails workstation:

![screen shot 2018-04-23 at 6 29 56 pm](https://user-images.githubusercontent.com/7832803/39161188-5bd543f4-4724-11e8-85f4-06d73b4df62d.png)

## Testing

1. Check out this branch in your Tails VM.
2. `./securedrop-admin tailsconfig`
3. The GUI will pop up at this stage because you are not running on the latest production tag. Ignore this (hit "Update Later") and reboot for the full experience.
4. Log in and enable an admin password
5. The updater should pop up. Attempt to upgrade your workstation (being intentionally vague).

## Deployment

Users will need to re-run `./securedrop-admin tailsconfig` to use this. 

But from that point forward, `securedrop-admin update`, `securedrop-admin setup` and `securedrop-admin tailsconfig` will be automated by the GUI. 

## Checklist

### If you made changes to the Updater GUI

- [x] `make ci-lint` passes
- [x] `python3 test_gui.py` passes
